common/lib/xmodule/xmodule/js/src/html/edit.coffee

```diff
@@ -958,6 +970,30 @@ class @HTMLEditingDescriptor
     if data['src']
       data['src'] = rewriteStaticLinks(data['src'], '/static/', @base_asset_url)
 
+  openImageModal: () =>
+    img = $(@visualEditor.selection.getNode())
+    imgAttrs = {}
+    if img && img.is('img')
+      imgAttrs['src'] = img.attr('src')
```

Comment:
Nit: might be a little cleaner by setting it all at once?

```
imgAttrs = {
  src: img.attr('src'),
  alt: img.attr('alt'),
  ...
}
```

Reply:
Because of the baseAssetUrl stuff, I'm now initializing `imgAttr` like this:

```coffeescript
imgAttr =
  baseAssetUrl: @base_asset_url
```

So, I can't just initialize it in that if condition now. Unfortunately, the coffeescript compiler the platform is using doesn't seem to support the [object spread](http://coffeescript.org/#splats) syntax which would allow me to clean that code up. E.g.

```coffeescript
imgAttr =
  imgAttr...
  src: img.attr('src')
  ... etc ...
```

I get a syntax error if I try that.

Reply:
Ah, ok!